### ISSUE TITLE
Ran into "504 Gateway Time-out" error when predicting using byo container

### ISSUE THREAD

**Issue by myralucy** [awslabs/amazon-sagemaker-examples#279]
I followed this [scikit-learn example](https://github.com/awslabs/amazon-sagemaker-examples/tree/master/advanced_functionality/scikit_bring_your_own/container), it works. 
Then I changed the Dockerfile, train.py and predictor.py to implement my own DL algos using mxnet. I am using the local-test mode after building a docker. The train_local.sh works. I start the serve locally and try predict.sh, it kept showing infos as follows:
"< HTTP/1.1 100 Continue
* We are completely uploaded and fine
< HTTP/1.1 504 Gateway Time-out
< Server: nginx/1.10.3 (Ubuntu)
< Date: Tue, 12 Jun 2018 00:30:01 GMT
< Content-Type: text/html
< Content-Length: 192
< Connection: keep-alive
<
<html>
<head><title>504 Gateway Time-out</title></head>
<body bgcolor="white">
<center><h1>504 Gateway Time-out</h1></center>
<hr><center>nginx/1.10.3 (Ubuntu)</center>
</body>
</html>
* Connection #0 to host localhost left intact"

I've been stuck by this error and have been debugging it for several days. I am pretty sure it has something to do with the line "import mxnet". As long as I remove this line, it can get response and at least print some logs rather than hang there with 504 Error.
THX!

**Comment by myralucy** [comment]
I have been debugging it for several days. I am pretty sure it is stuck in the line "import mxnet". As long as I remove this line, it can get response and at least print some logs rather than hanging there with 504 Error.

**Comment by djarpin** [comment]
Hi @myralucy .  Have you considered using SageMaker's pre-built MXNet container?  There is a built-in local mode with the Python SDK that pulls the pre-built MXNet container into your local environment.  That may be an easier path.  Here's an [example notebook](https://github.com/awslabs/amazon-sagemaker-examples/blob/master/sagemaker-python-sdk/mxnet_gluon_mnist/mxnet_mnist_with_gluon_local_mode.ipynb) showing how to use that.

**Comment by myralucy** [comment]
@djarpin Sure. I've tried using pre-built MXNet. 
But how can I get one docker image using pre-built MXNet? I've no idea, and that's why I was trying the example above.

**Comment by djarpin** [comment]
Closing due to lack of activity.  Feel free to re-open as necessary.  Thanks.

**Comment by geek311** [comment]
Having similar issue using Tensorflow model

from keras.models import Sequential
from keras.layers import Dense, LSTM

#Build the LSTM model
model = Sequential()
model.add(LSTM(50, return_sequences=True, input_shape= (x_train.shape[1], 1)))
model.add(LSTM(50, return_sequences= False))
model.add(Dense(25))
model.add(Dense(1))

# Compile the model
model.compile(optimizer='adam', loss='mean_squared_error')
#Train the model
model.fit(x_train, y_train, batch_size=1, epochs=1)

The above is where the Sagemaker notebook hangs and crashes with Gateway Timeout error. Using ml.t2.medium and 20 GB memory

**Comment by geek311** [comment]
Increased instance size for Sagemaker to xlarge and increased memory and it worked.

**Comment by barronh** [comment]
Having a similar issue. This seems to consistently happen when allocated memory resources exceed those requested in the instance. As far as I can tell, it crashes the hub and maybe the server. If so, it would be better for subprocess to be throttled to prevent the system going down.

Thoughts?
